numba/unicode.py

```diff
@@ -702,6 +702,41 @@ def _strncpy(dst, dst_offset, src, src_offset, n):
                             _get_code_point(src, src_offset + i))
 
 
+@intrinsic
+def _get_str_slice_view(typingctx, src_t, start_t, length_t):
+    """Create a slice of a unicode string using a view of its data to avoid
+    extra allocation.
+    """
+    assert src_t == types.unicode_type
+
+    def codegen(context, builder, sig, args):
+        src, start, length = args
+        in_str = cgutils.create_struct_proxy(
+            types.unicode_type)(context, builder, value=src)
+        view_str = cgutils.create_struct_proxy(
+            types.unicode_type)(context, builder)
+        view_str.meminfo = in_str.meminfo
+        view_str.kind = in_str.kind
+        view_str.length = length
+        # hash value -1 to indicate "need to compute hash"
+        view_str.hash = context.get_constant(_Py_hash_t, -1)
+        # get a pointer to start of slice data
+        byte_width = context.compile_internal(
```

Comment:
Does the `resolve_value_type`, `get_call_type` etc pattern not work here?

Reply:
Good point. This seemed fragile to me too. Fixed.